wolfcrypt/src/ed448.c

```diff
@@ -221,31 +271,29 @@ static int ed448_sign_msg(const byte* in, word32 inLen, byte* out,
 
         /* step 3: hash R + public key + message getting H(R,A,M) then
            creating S = (r + H(R,A,M)a) mod l */
-        ret = wc_InitShake256(&sha, NULL, INVALID_DEVID);
         if (ret == 0) {
-            ret = wc_Shake256_Update(&sha, ed448Ctx, ED448CTX_SIZE);
+            ret = ed448_hash_update(key, ed448Ctx, ED448CTX_SIZE);
             if (ret == 0) {
-                ret = wc_Shake256_Update(&sha, &type, sizeof(type));
+                ret = ed448_hash_update(key, &type, sizeof(type));
             }
             if (ret == 0) {
-                ret = wc_Shake256_Update(&sha, &contextLen, sizeof(contextLen));
+                ret = ed448_hash_update(key, &contextLen, sizeof(contextLen));
```

Comment:
Above `ret == 0` checked twice. Please remove 274 check, since it is done above.

Reply:
done